From: Oswald Buddenhagen <oswald.buddenhagen@qt.io>
Date: Thu, 19 Jan 2017 20:54:41 +0000 (+0100)
Subject: utilize configure results better in native builds
X-Git-Url: https://codereview.qt-project.org/gitweb?p=qt%2Fqtbase.git;a=commitdiff_plain;h=c458a197651b081ea780f560e8c50e0805aaffc2;hp=9ae3c94690e9a2579facc6c2827d4218422888ff

utilize configure results better in native builds

don't ignore detected features for host tools when we're not actually
cross-building.

Change-Id: Id62a3c1c6b7ae422b14efb4fbea0892b05a047cc
---

diff --git a/qtbase/mkspecs/features/default_post.prf b/qtbase/mkspecs/features/default_post.prf
index e7e9a5b..7f5ab3a 100644
--- a/qtbase/mkspecs/features/default_post.prf
+++ b/qtbase/mkspecs/features/default_post.prf
@@ -60,7 +60,7 @@ debug {
 }
 
 # disable special linker flags for host builds (no proper test for host support yet)
-!host_build {
+!host_build|!cross_compile {
     use_gold_linker: QMAKE_LFLAGS += $$QMAKE_LFLAGS_USE_GOLD
     enable_new_dtags: QMAKE_LFLAGS += $$QMAKE_LFLAGS_NEW_DTAGS
 }
diff --git a/qtbase/mkspecs/features/qt_common.prf b/qtbase/mkspecs/features/qt_common.prf
index 7967697..2dfc22f 100644
--- a/qtbase/mkspecs/features/qt_common.prf
+++ b/qtbase/mkspecs/features/qt_common.prf
@@ -17,7 +17,7 @@ qtConfig(c++14): CONFIG += c++14
 qtConfig(c++1z): CONFIG += c++1z
 contains(TEMPLATE, .*lib) {
     # module and plugins
-    !host_build:qtConfig(reduce_exports): CONFIG += hide_symbols
+    if(!host_build|!cross_compile):qtConfig(reduce_exports): CONFIG += hide_symbols
     unix:qtConfig(reduce_relocations): CONFIG += bsymbolic_functions
     qtConfig(separate_debug_info): CONFIG += separate_debug_info
 
